src/test/java/hudson/plugins/sonar/SonarInstallationTest.java

```diff
@@ -43,29 +50,37 @@
 public class SonarInstallationTest extends SonarTestCase {
 
   @Test
-  public void testRoundtrip() throws IOException {
+  public void testRoundtrip() throws IOException, ExecutionException, InterruptedException {
     TriggersConfig triggers = new TriggersConfig();
     SonarGlobalConfiguration d = new SonarGlobalConfiguration();
+    String credentialsId = "credentialsId";
     SonarInstallation inst = spy(new SonarInstallation(
       "Name",
       "server.url",
-      "credentialsId",
+            credentialsId,
       null,
       "secretId",
       "mojoVersion",
       "props",
       "key=value",
       triggers));
-    StringCredentials cred = new StringCredentialsImpl(CredentialsScope.GLOBAL, "an-id", null, Secret.fromString("token"));
-    doReturn(cred).when(inst).getCredentials(any(Run.class));
+    StringCredentials cred = new StringCredentialsImpl(CredentialsScope.GLOBAL, credentialsId, null, Secret.fromString("token"));
+
+    CredentialsStore store = CredentialsProvider.lookupStores(j.jenkins).iterator().next();
+    store.addCredentials(Domain.global(), cred);
+
+    FreeStyleProject freeStyleProject = j.createFreeStyleProject();
+    FreeStyleBuild run = freeStyleProject.scheduleBuild2(0).get();
+    j.waitForCompletion(run);
```

Comment:
```suggestion
    FreeStyleBuild run = new FreeStyleBuild(freeStyleProject);
```
There is no need to start it